### ISSUE TITLE
401 Unauthorized error on Fetch data page of React app with authentication after logging in (ASP.NET Core 3 Preview 3)

### ISSUE THREAD

**Issue by loop-evgeny** [aspnet/AspNetCore#8405]
### Describe the bug
After following the various workarounds I can actually register and account on my React app with authentication. However, when I browse to the Fetch data page loading the data fails with 401 Unauthorized.

### To Reproduce
1. Using ASP.NET Core 3.0 Preview 3 create a new project with dotnet new react --auth Individual
2. Apply the various workarounds at https://github.com/dotnet/core/blob/master/release-notes/3.0/preview/3.0.0-preview-known-issues.md
3. Run the app.
4. Click Register and complete the registration form.
5. Observe that the header shows "Hello youremail@..." so you are now logged in.
6. Click Fetch data in the header.

The page says "Loading..." and the developer console shows 2 JavaScript errors:

```
FetchData.js:4 GET https://localhost:44341/api/SampleData/WeatherForecasts 401 (Unauthorized)
_callee$ @ FetchData.js:4
tryCatch @ runtime.js:62
invoke @ runtime.js:288
prototype.(anonymous function) @ runtime.js:114
asyncGeneratorStep @ asyncToGenerator.js:3
_next @ asyncToGenerator.js:25
(anonymous) @ asyncToGenerator.js:32
(anonymous) @ asyncToGenerator.js:21
populateWeatherData @ FetchData.js:4
componentDidMount @ FetchData.js:13
commitLifeCycles @ react-dom.development.js:18071
commitAllLifeCycles @ react-dom.development.js:19630
callCallback @ react-dom.development.js:147
invokeGuardedCallbackDev @ react-dom.development.js:196
invokeGuardedCallback @ react-dom.development.js:250
commitRoot @ react-dom.development.js:19854
(anonymous) @ react-dom.development.js:21402
unstable_runWithPriority @ scheduler.development.js:255
completeRoot @ react-dom.development.js:21401
performWorkOnRoot @ react-dom.development.js:21324
performWork @ react-dom.development.js:21229
performSyncWork @ react-dom.development.js:21203
interactiveUpdates$1 @ react-dom.development.js:21488
interactiveUpdates @ react-dom.development.js:2268
dispatchInteractiveEvent @ react-dom.development.js:5086
```

```
Uncaught (in promise) SyntaxError: Unexpected end of JSON input
Promise.then (async)
asyncGeneratorStep @ asyncToGenerator.js:13
_next @ asyncToGenerator.js:25
Promise.then (async)
asyncGeneratorStep @ asyncToGenerator.js:13
_next @ asyncToGenerator.js:25
(anonymous) @ asyncToGenerator.js:32
(anonymous) @ asyncToGenerator.js:21
populateWeatherData @ FetchData.js:4
componentDidMount @ FetchData.js:13
commitLifeCycles @ react-dom.development.js:18071
commitAllLifeCycles @ react-dom.development.js:19630
callCallback @ react-dom.development.js:147
invokeGuardedCallbackDev @ react-dom.development.js:196
invokeGuardedCallback @ react-dom.development.js:250
commitRoot @ react-dom.development.js:19854
(anonymous) @ react-dom.development.js:21402
unstable_runWithPriority @ scheduler.development.js:255
completeRoot @ react-dom.development.js:21401
performWorkOnRoot @ react-dom.development.js:21324
performWork @ react-dom.development.js:21229
performSyncWork @ react-dom.development.js:21203
interactiveUpdates$1 @ react-dom.development.js:21488
interactiveUpdates @ react-dom.development.js:2268
dispatchInteractiveEvent @ react-dom.development.js:5086
```

### Expected behavior
The weather forecast data is shown.

### Screenshots
![image](https://user-images.githubusercontent.com/34301024/54139383-b56deb00-4421-11e9-8979-ebd997b28cc4.png)

**Comment by loop-evgeny** [comment]
I can see the following header in the network request (in Chrome dev tools):

`Authorization: Bearer [object Promise]`

**Comment by loop-evgeny** [comment]
The fix seems to be to add `await` to the line `const token = authService.getAccessToken();` in populateWeatherData() in FetchData.js:

`const token = await authService.getAccessToken();`

**Comment by marquesm91** [comment]
Fix in #8592
